cpp/include/raft/stats/histogram.cuh

```diff
@@ -61,6 +62,44 @@ void histogram(HistType type,
   detail::histogram<DataT, IdxT, BinnerOp>(type, bins, nbins, data, nrows, ncols, stream, binner);
 }
 
+/**
+ * @brief Perform histogram on the input data. It chooses the right load size
+ * based on the input data vector length. It also supports large-bin cases
+ * using a specialized smem-based hashing technique.
+ * @tparam value_t input data type
+ * @tparam idx_t data type used to compute indices
+ * @tparam binner_op takes the input data and computes its bin index
+ * @param[in]  handle the raft handle
+ * @param[in]  type histogram implementation type to choose
+ * @param[in]  data input data col-major (length = nrows * ncols)
+ * @param[out] bins the output bins col-major (length = nbins * ncols)
+ * @param[in]  binner the operation that computes the bin index of the input data
+ *
+ * @note signature of binner_op is `int func(value_t, IdxT);`
+ */
+template <typename value_t,
+          typename idx_t     = int,
```

Comment:
We should consider removing the default value_t and idx_t in all of these functions.